stix_shifter_modules/nozomi_vantage/stix_transmission/connector.py

```diff
@@ -55,31 +57,23 @@ async def create_results_connection(self, query, offset, length, metadata=None):
                     continue
 
                 return_obj['success'] = True
-                # remaining records value is less than fetched records.
-                if remaining_record_count and remaining_record_count < len(response_dict['result']):
-                    data += response_dict['result'][:remaining_record_count]
-                    result_count += len(response_dict['result'][:remaining_record_count])
-                else:
-                    data += response_dict['result']
-                    result_count += len(response_dict['result'])
-
-                if len(response_dict['result']) < page_size:
-                    page_number = None
-                    break
+                data += response_dict['result']
+                local_result_count = len(response_dict['result'])
+                result_count += local_result_count
 
-                # calculating remaining records count
-                if result_count < total_records:
-                    remaining_record_count = total_records - result_count
-                else:   # if doesn't have remaining records
+                if len(response_dict['result']) < self.api_client.api_page_size:
+                    page_number = None
                     break
 
-                page_number += 1
+                # Incrementing the next page when the fetched results are fully utilized.
+                if local_result_count <= expected_result_count:
+                    page_number += 1
 
-            return_obj = self.handle_data(data, metadata, offset, total_records, return_obj)
+            return_obj = self.handle_data(data, offset, expected_result_count, return_obj)
```

Comment:
How are you handling partial pages.

For example, 
Page size = 100
length = 101

What happens when it tries to get that remaining result?

Reply:
For example,
consider offset=0, length=101, pagesize=100

It will make two API calls.
From the first page, will fetch 100 records.
From the second page, will fetch 100 records.
from the 200 records first 101 records will be sliced.
Totally, it will return 101 records.

Reply:
What happens with the meta data in this case. What will the next query be when it goes to get the next 101 records.

Reply:
Updated the pagination as per the suggestion.